### ISSUE TITLE
Naming convention relaxedCamelCase does not allow consecutive upper case at end of identifier (like tRelaxedCC)

### ISSUE THREAD

**Issue by 456Xander** [jeremiah-c-leary/vhdl-style-guide#1548]
**Environment**
* vsg 3.35.0
* python 3.14.4
* OS=archlinux

**Describe the bug**

The naming convention relaxedCamelCase is described as "Enforce camelCase with any number of uppercase characters in a row".

However, multiple uppercase characters at the end of the identifier are not allowed.
We typically use this for camel case identifiers (types) that end with an acronym.

A minimum viable example is included here.

**To Reproduce**

VHDL code

```vhd
package repro is

  subtype tRelaxedCC is integer range 0 to 7;

end package repro;
```

VSG configuration:

```yaml
rule:
  subtype_501:
    case: "relaxedCamelCase"
```

run

```bash
vsg -c vsg.yaml -- repro.vhd
```

**Expected behavior**
No errors found, as tRelaxedCC is a relaxedCamelCase identifier.

**Screenshots**

Incorrect command output:

```
================================================================================
File:  repro.vhd
================================================================================
Phase 6 of 7... Reporting
Total Rules Checked: 886
Total Violations:    1
  Error   :     1
  Warning :     0
--------------+------------+------------+--------------------------------------
  Rule        |  severity  |  line(s)   | Solution
--------------+------------+------------+--------------------------------------
  subtype_501 | Error      |          3 | Format tRelaxedCC into relaxedCamelCase
--------------+------------+------------+--------------------------------------
NOTE: Refer to online documentation at https://vhdl-style-guide.readthedocs.io/en/latest/index.html for more information.
```

**Additional context**

**Comment by JHertz5** [comment]
Hi @456Xander. Thanks for raising this issue. The reason that this isn't working is because of the lack of a lower case character at the end of the string. The regex in use for relaxed camel case is as follows
`"(?:[a-z])+(?:[a-z0-9])*((?:[A-Z])+(?:[a-z0-9])+)*([A-Z])?"`. I can take a look at modifying the regex to allow the string to end with an uppercase letter.

**Comment by JHertz5** [comment]
@456Xander I've created a PR that I believe should resolve your issue. Could you please give that branch a try and confirm whether it solves your problem?

**Comment by 456Xander** [comment]
Sorry for the long wait, I was busy with something else and completely forgot about this issue.

Just tested the branch and it solves my problem. Thanks a lot.
